src/els_stdio.erl

```diff
@@ -42,7 +42,7 @@ loop(Lines, IoDevice, Server, JsonOpts) ->
       %% Use file:read/2 since it reads bytes
       {ok, Payload} = file:read(IoDevice, Length),
       Request       = jsx:decode(Payload, JsonOpts),
-      els_server:process_requests(Server, [Request]),
+      gen_server:cast(Server, {process_requests, [Request]}),
```

Comment:
Well, this was reeeeally confusing (and I think it could have something to do with the occasional timeouts we see in CI). We were calling:

```
els_server:process_requests(Server, [Request]),
```

But this function is actually called in two separate scenarios in the case of a stdio server: sometimes it calls the `els_server` (when sending requests), sometimes the client (when sending responses). In the latter case, the `els_client` implements its own `handle_cast` to handle this request. These two things are too tightly coupled and I plan to restructure this part. For now, I'm making sure we don't call `els_server` but we make an explicit cast to lower confusion.

Reply:
Nice! Doesn't the same apply in `els_tcp`, though? I think we are actually only using TCP connections in the proper testing.

Reply:
In `els_client`, there is a special function clause for stdio (see `handle_cast/2`). The requests over tcp are simply dropped.

Reply:
I guess the current approach is a little hacking, my bad 😿 .

Reply:
No worries! Your contribution actually allowed us to integrate with clients not supporting TCP. We can still clean things up if necessary!